js/reference.js

```diff
@@ -2769,189 +2828,60 @@ var ERMrest = (function(module) {
      * @memberof ERMrest
      * @constructor
      * @param {ERMrest.Reference} reference column's reference
-     * @param {?ERMrest.Column} base The column that this reference-column will be created based on.
-     * @param {?Object} kwargs if it's not empty then the column is being created based on foreignkey or key.
+     * @param {ERMrest.Column[]} baseCols List of columns that this reference-column will be created based on.
      * @desc
      * Constructor for ReferenceColumn. This class is a wrapper for {@link ERMrest.Column}.
      */
-    function ReferenceColumn(reference, base, kwargs) {
-
+    function ReferenceColumn(reference, cols) {
```

Comment:
Yes, you're right. At first it was supposed to be for a single column. But in case of ForeignKey and Key, there might be multiple columns. What I was doing before was using just the first column so that it's still just a single column. 

But I changed it to be an array. This way it will be more useful. There are some attributes that don't need to be overridden for case of multiple columns. For instance the `nullok` or `default` attribute. If the `ReferenceColumn` was just wrapper for one Column, then the `ForeignKeyPseudoColumn` and `KeyPseudoColumn` had to implement the same logic just for multiple columns.